src/gax.ts

```diff
@@ -67,27 +67,53 @@ import {toLowerCamelCase} from './util';
 
 /**
  * Per-call configurable settings for retrying upon transient failure.
+ * @implements {RetryOptionsType}
  * @typedef {Object} RetryOptions
- * @property {String[]} retryCodes
+ * @property {String[] | (function)} retryCodesOrShouldRetryFn
  * @property {BackoffSettings} backoffSettings
+ * @property {(function)} getResumptionRequestFn
  */
 export class RetryOptions {
-  retryCodes: number[];
+  retryCodesOrShouldRetryFn: number[] | ((error: any) => boolean);
   backoffSettings: BackoffSettings;
-  constructor(retryCodes: number[], backoffSettings: BackoffSettings) {
-    this.retryCodes = retryCodes;
+  getResumptionRequestFn?: (response: any) => any;
+  constructor(
+    retryCodesOrShouldRetryFn: number[] | ((error: any) => boolean),
+    backoffSettings: BackoffSettings,
+    getResumptionRequestFn?: (response: any) => any
+  ) {
+    this.retryCodesOrShouldRetryFn = retryCodesOrShouldRetryFn;
     this.backoffSettings = backoffSettings;
+    this.getResumptionRequestFn = getResumptionRequestFn;
   }
 }
 
+/**
+ * Per-call configurable settings for working with retry-request
+ * See the repo README for more about the parameters
+ * https://github.com/googleapis/retry-request
+ * Will be deprecated in a future release. Only relevant to server streaming calls
+ * @typedef {Object} RetryOptions
+ * @property {boolean} objectMode - when true utilizes object mode in streams
+ * @property {request} request - the request to retry
+ * @property {number} noResponseRetries - number of times to retry on no response
+ * @property {number} currentRetryAttempt - what # retry attempt retry-request is on
+ * @property {Function} shouldRetryFn - determines whether to retry, returns a boolean
+ * @property {number} maxRetryDelay - maximum retry delay in seconds
+ * @property {number} retryDelayMultiplier - multiplier to increase the delay in between completion of failed requests
+ * @property {number} totalTimeout - total timeout in seconds
+ */
 export interface RetryRequestOptions {
   objectMode?: boolean;
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   request?: any;
   retries?: number;
   noResponseRetries?: number;
   currentRetryAttempt?: number;
-  shouldRetryFn?: () => boolean;
+  shouldRetryFn?: (error: any) => boolean;
```

Comment:
This is formally a breaking change. We can avoid breaking change by making `error` parameter optional, if it works for us.

Reply:
Good to know. Question + additional context - Is it specifically line 113 or is it because of the lines below as well? I added these parameters to the interface because they're legal parameters for `retry-request` and while the version of `retry-request` had been bumped in this library, this interface had not been updated to correspond with the parameters accepted in the most recent version

Reply:
Actually, my bad: apparently, adding a parameter in this case is not considered a breaking change in TypeScript ([playground link](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgPJWAcwEoTFAT1wEcBXCAZzFQAcxgB7EC5AbwChkvkKALB0gBsAJrnwEAYiABcyABQBKZAF4AfMgBGDBoIhwQAbnYBfduwRMqyAO4MoAawqz0WMYRLkqtepZVtO3HwCIm6SMvJKamzIUHikUCDI8IIUEAbIpsbIZqCQsIgoAHIQ1qEelNR0jMz+3Dz8QqJ4hFKyctBQsgCiUFB2kepaOnqGJmYWzGA89IKCAOp2jrLFpc1EEGQV3tUsyrWBDSFrrREq6qwxcQlJcClpGWNAA)).

I would still recommend to say that `error` is optional.

Reply:
Done!